apps/cave/components/CurrencyAmountButton/CurrencyAmountButton.tsx

```diff
@@ -52,8 +51,8 @@ export const useCurrencyButtonState = (
     if (!currency) return 'no currency'
     if (currency.isNative) return 'successful'
     if (approve.isError && approve.error['code'] !== 4001) return 'error'
-    if (totalSupply.greaterThan(0) && approve.allowance?.amount?.greaterThan(totalSupply))
-      return 'successful'
+    // if (totalSupply.greaterThan(0) && approve.allowance?.amount?.greaterThan(totalSupply))
```

Comment:
check line 20

```ts
const approve = useApprove(currency?.wrapped, spender)
```

in this case, you will have to approve once for each transaction because the default value is `MaxUint256.toString()`